### DISCUSSION TITLE
有没有办法固定外网的打洞端口

### DISCUSSION THREAD

**Discussion by zycbscn** [MikeWang000000/Natter#61] [Q&A]
你好，我主要要解决没有IPV4公网IP，通过外网访问私有NAS。目前已部署成功，我是个小白，有两个问题请大佬帮看看能否解决。
1、NAS固定每天关机5小时，重启主机后，是否需要重新运行Natter？
2、每次打洞端口都会变，导致每次端口失效后需要进系统看端口情况，是否能将端口固定住，不用每次修改访问端口。
感谢大神指导。
顺便求助下问题，目前我也可以通过V6公网地址+域名转换访问主机。但会出现偶尔，APP访问不成功（网页访问一切正常）情况。之前以为是解析的问题，咨询了域名提供商，反馈解析没有问题。不知道大佬是否能指导下。感谢。

**Comment by MikeWang000000** [comment]
- 关于1：如果您的 Natter 运行在路由器而非 NAS 上，NAS 重启不影响打洞。
- 关于2：公网端口是运营商CGN服务器决定的，它会随机挑一个端口给您使用。Natter没有办法保证两次端口号相同（除非您运气好，随机到了同一个端口）
- 关于最后一个问题：建议咨询 App 客服或者开发者，大概是 App 没有做IPv6适配。

**Comment by zycbscn** [comment]
感谢大佬解答。



| |
zy
|
|
***@***.***
|




---- 回复的原邮件 ----
| 发件人 | Mike ***@***.***> |
| 日期 | 2023年03月29日 22:27 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [MikeWang000000/Natter] 有没有办法固定外网的打洞端口 (Issue #27) |

关于1：如果您的 Natter 运行在路由器而非 NAS 上，NAS 重启不影响打洞。
关于2：公网端口是运营商CGN服务器决定的，它会随机挑一个端口给您使用。Natter没有办法保证两次端口号相同（除非您运气好，随机到了同一个端口）
关于最后一个问题：建议咨询 App 客服或者开发者，大概是 App 没有做IPv6适配。
